### PR TITLE
do not recompute lf if already existing

### PR DESCRIPTION
**Pull request by mcpiastra**
this relates to the issue #2188 
after the dipole scan, the leadfield should be recomputed only if not already existing

### CODE REVIEW

**Review by schoffelen — COMMENTED** [review]
Hmmm, are you sure that this change will work? Also in general?

Specifically, you check for ~lf, but this will throw an error if the variable lf does not exist in the function. This could be the case for instance if someone runs a nonlinear fit only (i.e. without gridsearch).
Also, generically, lf to me seems the forward solution to one of the dipoles in the grid, which is updated in the for-loop across the dipoles in the grid. The lf outside the gridsearch loop is just the leadfield of the last dipole in the grid, and not necessarily the one that gave the optimal solution.

Or am I missing something?

**Comment by mcpiastra** [comment]
you are right, sorry.
I now modified the code so that the corresponding leadfield is copied along the dipole, and there is a check on the field of of dipole.
does it make sense to you now?
thanks.

File: ft_dipolefitting.m [hunk 1]
```diff
@@ -590,8 +596,12 @@
   case 'moving'
     for t=1:ntime
       if success(t)
-        % re-compute the leadfield in order to compute the model potential and dipole moment
-        lf = ft_compute_leadfield(dip(t).pos, sens, headmodel, leadfieldopt{:});
+        if ~isfield(dip, 'lf')  
+            % if there is no leadfield, re-compute it in order to compute the model potential and dipole moment
+            lf = ft_compute_leadfield(dip(t).pos, sens, headmodel, leadfieldopt{:});
+        else
+          lf = dip.lf;
```
Comment (schoffelen) [inline ft_dipolefitting.m:603]: would you need to consider the loop-variable (and also in line 466?)?

**Comment by schoffelen** [comment]
Hi @mcpiastra, have you tried to run the suggested test functions on your end (given the changed code)? Let me know if those functions ran through without error, then I will merge.

**Comment by mcpiastra** [comment]
hi @schoffelen, after the last commit, yes, all the suggested tests run with no error.
thanks!

**Comment by schoffelen** [comment]
Hi @mcpiastra what needs to be done to close/merge this?

**Comment by mcpiastra** [comment]
Hi @schoffelen , actually I was recently having the same question. What shall I do? Write a short script where this is used, with a dummy leadfield?

**Comment by schoffelen** [comment]
Well, indeed it would be good to have a snippet of code (either in an appropriate already existing test function, or in a new test function), that just goes through the movements.
This would mean a call to ft_dipolefitting with a precomputed leadfield, with gridsearch = 'yes', and without the nonlinear optimization step. If it runs through without error, you're done.

**Pull request merged**